### PR TITLE
Etk 17 buttons

### PR DESCRIPTION
**Pull request by devicedev**
Add new button component

### CODE REVIEW

File: src/Button/Button.jsx [hunk 1]
```diff
@@ -1,27 +1,45 @@
...
+    }
+  }
+  let spinnerColor = spinnerVariants[variant]
+  if (loading && disabled && (variant === 'primary' || variant === 'large')) {
+    spinnerColor = {
+      background: colors.disabled,
+      primary: colors.white
+    }
+  }
+  console.log(colors)
```
Comment (alsofronie) [inline src/Button/Button.jsx:41]: no debug info on PR code, please

File: src/Button/Button.jsx [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
+import { Flex, Space } from '../'
+import variants, { iconVariants, spinnerVariants } from './variants'
+import { Icon } from '../Icon'
+import colors from '../assets/colors'
 
 const Button = ({
   children,
   disabled,
-  icon,
+  iconFace,
```
Comment (alsofronie) [inline src/Button/Button.jsx:17]: Isn't this a breaking change? Why can't you use the `icon` prop?

File: src/Button/Button.jsx [hunk 3]
```diff
@@ -30,57 +48,36 @@ const Button = ({
...
+          px={3}
+          color='inherit'>
+          {children}
+        </Typography>
+        {(iconFace && iconPosition === 'right') &&
+        <Space ml={'10px'}>
+          <Icon name={iconFace} size={iconVariants[variant].fontSize} color={iconColor}/>
+        </Space>
+        }
+      </Flex>
```
Comment (alsofronie) [inline src/Button/Button.jsx:75]: The loading icon replaces the present icon, it does not add a new one.

File: stories/Button.stories.jsx [hunk 4]
```diff
@@ -12,12 +12,13 @@ export default {
...
-    'outlineAlt',
-    'positive',
+    'large',
     'link',
     'linkSecondary',
-    'linkPositive'
+    'linkPositive',
+    'outline',
+    'outlineAlt',
+    'positive'
```
Comment (alsofronie) [inline stories/Button.stories.jsx:21]: I cannot find anywhere in the new styleguide the outline and outline alt variants. We should map these values (for backwards compatibility) with the `primary` style.

**Review by alsofronie — CHANGES_REQUESTED** [review]

File: stories/Button.stories.jsx (hunk 4 above)
Comment (devicedev) [reply stories/Button.stories.jsx:21]: You mean to map all these: 
    'linkSecondary',
    'linkPositive',
    'outline',
    'outlineAlt',
    'positive'
with the primary style?

**Review by alsofronie — CHANGES_REQUESTED** [review]

**Review by alsofronie — CHANGES_REQUESTED** [review]

File: src/Button/Button.jsx [hunk 5]
```diff
@@ -1,27 +1,44 @@
...
+                  disabled,
+                  icon,
+                  id,
+                  loading,
+                  onClick,
+                  type,
+                  variant,
+                  iconPosition,
+                  ...rest
+                }) => {
```
Comment (alsofronie) [inline src/Button/Button.jsx:25]: Please use the prettier/webstorm config, this indentation is not according to our rules.

File: src/Button/Button.jsx [hunk 6]
```diff
@@ -30,57 +47,36 @@ const Button = ({
     {...rest}>
     {loading ? (
       <ActivityIndicator
-        colors={spinnerVariants[variant]}
+        colors={spinnerColor}
```
Comment (alsofronie) [inline src/Button/Button.jsx:50]: What if the button has the same background as the spinerColor? Shouldn't we use spinner variants here?

File: src/Button/variants.js [hunk 7]
```diff
@@ -1,67 +1,166 @@
...
+export const iconVariants = {
   primary: {
-    background: 'accent',
-    primary: 'white'
-  },
-  outline: {
-    background: 'white',
-    primary: 'accent'
+    color: colors.white,
+    fontSize: '1'
```
Comment (alsofronie) [inline src/Button/variants.js:127]: I think this is an index, we should use integer? Please check this:
```suggestion
    fontSize: 1
```

File: src/Button/variants.js [hunk 8]
```diff
@@ -1,67 +1,166 @@
...
-  outlineAlt: {
-    background: 'white',
-    primary: 'text'
-  },
-  positive: {
-    background: 'positive',
-    primary: 'white'
+  large: {
+    color: colors.white,
+    fontSize: '2'
```
Comment (alsofronie) [inline src/Button/variants.js:131]: same as before

File: src/Button/variants.js [hunk 9]
```diff
@@ -1,67 +1,166 @@
...
-  },
-  linkSecondary: {
-    background: 'white',
-    primary: 'text'
-  },
-  linkPositive: {
-    background: 'white',
-    primary: 'positive'
+    color: colors.brand,
+    fontSize: '1'
```
Comment (alsofronie) [inline src/Button/variants.js:135]: same as before

**Review by alsofronie — CHANGES_REQUESTED** [review]

File: src/Input/Input.jsx [hunk 10]
```diff
@@ -140,16 +113,12 @@ export const Input = forwardRef(
...
-                  fontSize={3}
-                  name={currentIcRight}
-                  color={currentIcRightColor}
-                  rotate={currentIcRight === 'loading'}
-                />
+                <Icon fontSize={3} name={icRight} />
               </Space>
-            ) : (
+            ) : null}
+            {type === 'password' && value ? (
```
Comment (alsofronie) [inline src/Input/Input.jsx:121]: Can't we have disabled password input?

Comment (devicedev) [reply src/Input/Input.jsx:121]: I merged with the changes from v1, I did not make myself this change.

File: src/assets/sizes.js [hunk 11]
```diff
@@ -1,3 +1,6 @@
-export default {
-  inputHeight: '40px'
-}
+export default {
+  inputHeight: '40px',
+  buttonMinWidth: '160px',
+  buttonMinHeight: '40px',
+  largeButtonMinWidth: '190px'
+}
```
Comment (alsofronie) [inline src/assets/sizes.js:6]: I saw in other frameworks a `controlHeight` property which is used for both input and normal buttons. Can't we refactor this like:
```suggestion

const BASE_SIZES = {
  controlHeight: '40px'
}

export default {
  ...BASE_SIZES,
  inputHeight: BASE_SIZES.controlHeight,
  buttonMinWidth: '160px',
  buttonMinHeight: BASE_SIZES.controlHeight,
  largeButtonMinWidth: '190px'
}
```

Comment (alsofronie) [reply src/assets/sizes.js:6]: Are you sure we have a button min width? Look [here](https://www.figma.com/file/7Wovqa84J9V3XSb6IzEUuK/IN-PROGRESS---Etvas-Styleguide?node-id=1058%3A115)

Comment (alsofronie) [reply src/assets/sizes.js:6]: Are you sure the button has a minHeight? Maybe it's buttonHeight?

**Review by alsofronie — APPROVED** [review]

**Pull request merged**